MovieApp/MovieCover.swift

```diff
@@ -0,0 +1,182 @@
+//
+//  MovieDetailsWithCoverView.swift
+//  MovieApp
+//
+//  Created by endava-bootcamp on 02.04.2023..
+//
+
+import Foundation
+import PureLayout
+import MovieAppData
+
+class MovieCover: UIView {
+    
+    private var poster: UIImageView!
+    private var userRatingNumber: UILabel!
+    private var userRating: UILabel!
+    private var year: UILabel!
+    private var date: UILabel!
+    private var category: MovieCategoryModel!
+    private var genre: UILabel!
+    private var movieDetails: MovieDetailsModel!
+    private var movieTitle: UILabel!
+    private var favourite: UIButton!
+    private var duration: UILabel!
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    init() {
+        super.init(frame: CGRect())
+        movieDetails = MovieUseCase().getDetails(id: 111161)
+        buildViews()
+    }
+    
+    private func buildViews(){
+        createViews()
+        setViews()
+        defineLayoutForViews()
+    }
+    
+    private func createViews() {
+        poster = UIImageView()
+        movieTitle = UILabel()
+        year = UILabel()
+        favourite = UIButton()
+        userRating = UILabel()
+        userRatingNumber = UILabel()
+        duration = UILabel()
+        genre = UILabel()
+        date = UILabel()
+    }
+    
+    private func setViews() {
+        
+        //poster - movie cover image
+        downloadImage(from: URL(string: movieDetails.imageUrl)!)
```

Comment:
Why do you use force unwrapping here? You should avoid using force unwrapping and use optional binding.